pkg/graphviz/traceflow.go

```diff
@@ -341,6 +350,63 @@ func GenGraph(tf *crdv1alpha1.Traceflow) (string, error) {
 		graph.Attrs[gographviz.Label] = getTraceflowStatusMessage(tf)
 	}
 	if tf == nil || senderRst == nil || tf.Status.Phase != crdv1alpha1.Succeeded || len(senderRst.Observations) == 0 {
+		// For live traffic, when the source is IP or empty, there is no result from the sender Node result in the traceflow status.
+		if senderRst == nil && tf.Spec.LiveTraffic && tf.Status.Phase == crdv1alpha1.Succeeded {
+			// Draw the nodes for the sender.
+			srcCluster, err := createClusterWithDefaultStyle(graph, clusterSrcName)
+			if err != nil {
+				return "", err
+			}
+			srcCluster.Attrs[gographviz.Label] = "source"
+			srcCluster.Attrs[gographviz.LabelJust] = "l"
+			// For live traffic data, we only know src IP from capturedPacket
+			node, err := createEndpointNodeWithDefaultStyle(graph, srcCluster.Name, getWrappedStr(tf.Status.CapturedPacket.SrcIP))
+			if err != nil {
+				return "", err
+			}
+
+			// create an invisiable edge before destination cluster, otherwise the source cluster will
+			// always be on the right even source subGraph is before desitination subGraph in graph string.
+			err = graph.AddEdge(node.Name, node.Name, true, map[string]string{
+				"style": "invis",
+			})
+			if err != nil {
+				return "", err
+			}
+			dstCluster, err := createClusterWithDefaultStyle(graph, clusterDstName)
+			if err != nil {
+				return "", err
+			}
+			nodes, err := genSubGraph(graph, dstCluster, receiverRst, &tf.Spec, getDstNodeName(tf), false, 0)
+			if err != nil {
+				return "", err
+			}
+			// Add captured packet info to destination cluster.
+			err = graph.AddNode(dstCluster.Name, "capturedPacket", map[string]string{
```

Comment:
why only for this case (`senderRst == nil`)?

Reply:
it's a mistake, I will refine it in a following PR.